File: internal/incrementalcompiler-core/src/main/scala/sbt/internal/inc/Lookup.scala
package sbt.internal.inc

import java.io.File

import xsbti.compile.CompileAnalysis

trait Lookup {
  def lookupOnClasspath(binaryClassName: String): Option[File]
  def lookupAnalysis(classFile: File): Option[CompileAnalysis]
  def lookupAnalysis(binaryDependency: File, binaryClassName: String): Option[CompileAnalysis]
  def lookupAnalysis(binaryClassName: String): Option[CompileAnalysis]
}

File: internal/compiler-interface/src/main/java/xsbti/compile/Lookup.java
package xsbti.compile;


import xsbti.Maybe;

import java.io.File;

public interface Lookup {
    Maybe<File> lookupOnClasspath(String binaryClassName);
    Maybe<CompileAnalysis> lookupAnalysis(File classFile);
    Maybe<CompileAnalysis> lookupAnalysis(File binaryDependency, String binaryClassName);
    Maybe<CompileAnalysis> lookupAnalysis(String binaryClassName);
}
